modules/tinymce/src/core/main/ts/textpatterns/keyboard/Keyboard.ts

```diff
@@ -9,6 +9,8 @@ const setup = (editor: Editor): void => {
   const charCodes = [ ',', '.', ';', ':', '!', '?' ];
   const keyCodes = [ 32 ];
 
+  const isSelectionCollapsed = editor.selection.isCollapsed();
```

Comment:
I don't think this is the right spot. This will just calculate if the selection is collapsed the first time it's setup. You want to do it every keydown event. So closer to line 28, or maybe line 24 to avoid the text pattern lookup.

Reply:
Or alternatively, this should be a function, and then it will be invoked elsewhere. At the moment, it's a value.

Reply:
But ideally, a test should fail because of this. So instead of just changing it, see if you can make a test that fails because it's in the wrong spot.

Reply:
I have fixed it. I didn't move and convert it correctly.

Reply:
FYI, `ReplacementTest` already covers that case. Its tests failed before when I had the isSelectionCollapsed checking in the wrong spot